components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java

```diff
@@ -2231,11 +2231,20 @@ public String createApplication(ServiceProvider application, String tenantDomain
         String resourceId = doAddApplication(application, tenantDomain, username, applicationDAO::addApplication);
 
         for (ApplicationResourceManagementListener listener : listeners) {
-            if (listener.isEnabled() && !listener.doPostCreateApplication(resourceId, application, tenantDomain,
-                    username)) {
+            try {
+                if (listener.isEnabled() && !listener.doPostCreateApplication(resourceId, application, tenantDomain,
+                        username)) {
+                    log.error("Post create application operation of listener:" + getName(listener) + " failed for " +
+                            "application: " + application.getApplicationName() + " of tenantDomain: " + tenantDomain);
+                    break;
+                }
+            } catch (Throwable e) {
```

Comment:
why do we need to handle throwable or expectation here? isn't it handle in the top layer?

Reply:
Since we don't have the applicationID get from this layer, The rollback will happens only for the inbound. Which will cause again for partial app creation. ( This was before as well for exception `IdentityApplicationManagementException`)